packages/yew-macro/src/props/prop.rs

```diff
@@ -74,14 +76,43 @@ impl Prop {
                 "expected an expression following this equals sign",
             ));
         }
-        let value = strip_braces(input.parse::<Expr>()?)?;
+
+        let value = parse_prop_value(input)?;
         Ok(Self { label, value })
     }
 }
 
-fn strip_braces(expr: Expr) -> syn::Result<Expr> {
-    match expr {
-        Expr::Block(ExprBlock { block: Block { mut stmts, .. }, .. }) if stmts.len() == 1 => {
+fn parse_prop_value(input: &ParseBuffer) -> syn::Result<Expr> {
+    if input.peek(Brace) {
+        strip_braces(input.parse()?)
+    } else if let Some(ExprRange {
+        from: Some(from), ..
+    }) = range_expression_peek(input)
+    {
+        // If a range expression is seen, treat the left-side expression as the value
+        // and leave the right-side expression to be parsed as a base expression
+        advance_until_next_dot2(input)?;
+        Ok(*from)
+    } else {
+        let expr = input.parse()?;
+        match &expr {
+                Expr::Lit(_) => Ok(expr),
+                _ => {
+                            Err(syn::Error::new_spanned(
+                            &expr,
+                            "the property value must be either a literal or enclosed in braces. Consider adding braces around your expression.".to_string(),
+                            ))
+                        }
+            }
```

Comment:
The formatting seems weird here 🤔

Reply:
Weird. Rustfmt doesn't seem to care. I'll fix this by hand

Reply:
Weird. Rustfmt doesn't seem to care. I'll fix this by hand.